### DISCUSSION TITLE
ChangedSectionIdentifier: This element is not expected.

### DISCUSSION THREAD

**Discussion by fabiodisconzi** [OP-TED/eForms-SDK#875] [Q&A] [answered]
Hi to all.
Validator of some notices of january 2024 gives me this error:

> Error 1871: Element '{http://data.europa.eu/p27/eforms-ubl-extension-basic-components/1}ChangedSectionIdentifier': This element is not expected. Expected is one of ( {http://data.europa.eu/p27/eforms-ubl-extension-basic-components/1}ChangeDescription, {http://data.europa.eu/p27/eforms-ubl-extension-basic-components/1}ProcurementDocumentsChangeDate, {http://data.europa.eu/p27/eforms-ubl-extension-basic-components/1}ProcurementDocumentsChangeIndicator, {http://data.europa.eu/p27/eforms-ubl-extension-aggregate-components/1}ChangedSection ).

I cannot solve it since the "ChangedSectionIdentifier" is present in EFORMS-ExtensionBasicComponents-2.3.xsd
`<xsd:element name="ChangedSectionIdentifier" type="ChangedSectionIdentifierType"/>`

Many thanks,
Fabio

**Comment by YvesJo** [comment, marked as answer]
Hi,
The problematic notices are improperly formed with efbc:ChangedSectionIdentifier having "efac:Change" instead of "efac:ChangedSection" as parent element:
![image](https://github.com/OP-TED/eForms-SDK/assets/42802091/b794ed56-d669-4030-9676-1c642d96ebd0)

**Comment by fabiodisconzi** [comment]
Many thanksYvesJo!

For example the following section from 00006245_2024.xml:
```
<efac:Change>
  <efbc:ChangedSectionIdentifier>CON-0037</efbc:ChangedSectionIdentifier>
  <efbc:ChangeDescription languageID="RON">actul aditional nr .88005/27.12.2023 - diminuare cu 2197.44 lei fara tva</efbc:ChangeDescription>
</efac:Change>
```
should be:
```
<efac:Change**Section**>
  <efbc:ChangedSectionIdentifier>CON-0037</efbc:ChangedSectionIdentifier>
  <efbc:ChangeDescription languageID="RON">actul aditional nr .88005/27.12.2023 - diminuare cu 2197.44 lei fara tva</efbc:ChangeDescription>
</efac:Change**Section**>
```

For my target (a web app made via php to visualize notices), this seems a minor issue.

I would like to point ou that **there are a lot of notice that not pass the validation** :-( (at least for archive that I analyze).
Just for the archive number 3 of 2004 of 04 january 2024 I find is suppose more than 50 affected notices. 
For instance:
00006143_2024.xml (many validating errors)
00006245_2024.xml
00006267_2024.xml
00006129_2024.xml
...

**Reply by bertrand-lorentz** [reply]
The schema was updated to what @YvesJo described in SDK 1.7.0 (see the [corresponding release notes](https://github.com/OP-TED/eForms-SDK/releases/tag/1.7.0))

The [first notice in your list](https://ted.europa.eu/en/notice/-/detail/6143-2024) was created with SDK 1.6, so it is valid against the schema in that version of the SDK.
I assume it's the same for the other notices.

Every notice XML file includes a `cbc:CustomizationID` element that indicates the version of the SDK that was used to create the notice. When you try to validate a notice, you must take into account this version. See the following page for more details:
https://docs.ted.europa.eu/eforms-common/versioning/index.html#_significance_of_the_sdk_version_in_notice_handling_and_validation

**Reply by YvesJo** [reply]
Hi,
It is important to make sure that the XML instance complies with the schema it has been created with.
I've downloaded the first file referenced (00006143_2024.xml) from TED and parsed it against the XSD of SDK 1.6 and couldn't find any error. Be aware that the structure of the Change section evolved between SDK 1.6 and SDK 1.7.
If you still encounter error while parsing, it could be worth to check that the correct XSD version is used and that the tool used is properly configured to do so.
For most recent SDKs, the structure should be something like:
`<efac:Change>
<efbc:ChangeDescription languageID="RON">actul aditional nr .88005/27.12.2023 - diminuare cu 2197.44 lei fara tva</efbc:ChangeDescription>
<efac:ChangedSection>
<efbc:ChangedSectionIdentifier>CON-0037</efbc:ChangedSectionIdentifier>
</efac:ChangedSection>
</efac:Change>`

**Reply by fabiodisconzi** [reply]
Many thanks to all.
Where can I find the folders and file:

**/common 
/maindoc
schemas.json**

in version 1.6? 
And in other versions?

So in my php application I can select which schemas use according to the value of element `cbc:CustomizationID`. 
Many thaks,
Fabio

**Reply by YvesJo** [reply]
Hi,
The files may be found in the schemas folder of the SDK version tagged 1.6.4.
For any SDK x.y version go to the version tagged x.y.z where z is the highest fix number.
KR,

**Reply by fabiodisconzi** [reply]
Many thanks for the answers.
Surely I lack some competencies in GIT but, **where can I find all the SDK versions?** Here (https://github.com/OP-TED/eForms-SDK/tree/develop/schemas ) I can only find the last version.
Thanks.

A few words on what I am trying to do.
Keeping in mind that I only want to do some data analysis of the tenders (not submit any), currently the logic that I am trying to implement is the following:
- via Xpath read the `/*/cbc:CustomizationID`
- select the correspondent scheme
- validate
- go ahead with other ideas (save company that wins, save CACs, etc.).

**Reply by YvesJo** [reply]
On the left dropdown "Develop" in the Tags tab select the needed tagged version
![image](https://github.com/OP-TED/eForms-SDK/assets/42802091/540733a7-f098-453e-93c8-8839d1f52151)

**Comment by ferrakaop** [comment]
On a side note, you might be interested in the TED reuser workshop next Wednesday 17/4 and also in previous workshops - see https://op.europa.eu/en/web/ted-reusers-workshops/home

**Comment by fabiodisconzi** [comment]
First of all, many thanks for the kind answers. This is giving me the commitment to (try to) create a new dashboard related to tenders.
I share with you the first step: now the validation seems to work.
Thanks.

![eforms-sdk-schema](https://github.com/OP-TED/eForms-SDK/assets/15421089/8fbaefed-8cfc-47f9-aa28-e09ee52a0643)
